src/common/components/giftViews/giftListItem/giftListItem.component.js

```diff
@@ -0,0 +1,31 @@
+import angular from 'angular';
+import template from './giftListItem.tpl';
+import desigSrc from 'common/directives/desigSrc.directive';
+
+let componentName = 'giftListItem';
+
+class GiftListItemController {
+
+  /* @ngInject */
+  constructor() {
+  }
+}
+
+export default angular
+  .module( componentName, [
+    template.name,
+    desigSrc.name
+  ] )
+  .component( componentName, {
+    controller:  GiftListItemController,
+    templateUrl: template.name,
+    transclude:  {
+      'selectInput': '?label'
+    },
+    bindings:    {
+      gift:        '<',
```

Comment:
The object is bound by reference. If you set `this.gifts = ['a']` in the component, the parent wouldn't see this since it's a new object. The parent would see this `this.gifts.push('a')` though, since it's modifying the existing object and they are bound by reference. This is the same for object properties. Angular isn't copying the object when it's passed to the component.

The difference between one-way, and two-way, is that two-way, the parent would see `this.gifts = ['a']'.

See [component-based-application-architecture](https://docs.angularjs.org/guide/component#component-based-application-architecture), the section starting with "Inputs should be using "